app/classifier.py

```diff
@@ -51,59 +49,66 @@ def _load_instance(self, path_to_model: str):
+            self.vocab = [line.rstrip() for line in f.readlines() if line.rstrip() != '']
 
-    def classify(self, seq: str, model_name: str) -> [(str, float)]:
-        """ classify calculates and returns a particular sequence's topic probability vector.
+    def classify(self, seqs: List[str], model_name: str, fixed_threshold: Optional[float] = None) \
+            -> Dict[str, List[Tuple[str, float]]]:
+        """ classify calculates and returns all the sequences' topic probability vectors.
 
         Parameters:
-            seq (str): The text sequence to be classified with this model.
-            model_name (str): the name of the training model (e.g.
-                    UPR_2percent_ps0)
+            seqs ([str]): The text sequences to be classified with this model.
+            model_name (str): the name of the training model (e.g. UPR_2percent_ps0).
+            fixed_threshold (float or None): if set, apply the same threshold to all topics.
 
         Returns:
-            [(str, float)]: The topic probabilty vector in descending order,
-                    with topics below the minimum threshold (default=0.4)
+            {str: [(str, float)]}: Per sequence, the topic probabilty vector in descending
+                    order, with topics below the minimum threshold (default=topic-dependent)
                     discarded.
         """
+        if len(seqs) == 0:
+            return {}
 
         model_config_path = os.path.join(self.base_classifier_dir, model_name)
         self._load_instance(model_config_path)
 
         instance_dir = os.path.join(model_config_path, self.instance)
         embedder = Embedder(self.instance_config.bert)
         self._load_vocab(os.path.join(
-                instance_dir,
-                self.instance_config.vocab))
+            instance_dir,
+            self.instance_config.vocab))
 
         predictor = tf.contrib.predictor.from_saved_model(instance_dir)
-        embedding = embedder.GetEmbedding(seq)
-        input_mask = [1] * embedding.shape[0]
+        embeddings = embedder.GetEmbedding(seqs)
+        embedding_shape = embeddings[seqs[0]].shape
+        all_embeddings = np.zeros([len(embeddings), MAX_SEQ_LENGTH, embedding_shape[1]])
+        all_input_mask = np.zeros([len(embeddings), MAX_SEQ_LENGTH])
+
+        for i, (_, matrix) in enumerate(embeddings.items()):
+            all_embeddings[i][:len(matrix)] = matrix
+            all_input_mask[i][:len(matrix)] = 1
 
-        # classify seq, with its embedding matrix, using a specific model
         features = {
-            "embeddings": np.expand_dims(embedding, axis=0),
-            "input_mask": np.expand_dims(input_mask, axis=0),
+            "embeddings": all_embeddings,
+            "input_mask": all_input_mask,
         }
 
         predictions = predictor(features)
-        probabilities = predictions["probabilities"][0]
+        probabilities = predictions["probabilities"]
         logging.getLogger().debug(probabilities)
```

Comment:
you're not touching this but the getLogger() call should probably go in __init__ not in the classify() call.